plugins/flutter_aepmessaging/android/src/main/kotlin/com/adobe/marketing/mobile/flutter/flutter_aepmessaging/FlutterAEPMessagingPlugin.kt

```diff
@@ -0,0 +1,100 @@
+package com.adobe.marketing.mobile.flutter.flutter_aepmessaging
+
+import FlutterAEPMessagingDelegate
+import com.adobe.marketing.mobile.*
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+
+
+/** FlutterAEPMessagingPlugin */
+class FlutterAEPMessagingPlugin : FlutterPlugin, MethodCallHandler {
+  private lateinit var channel: MethodChannel
+  private var messageCache = mutableMapOf<String, Message>();
+
+  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "flutter_aepmessaging")
+    channel.setMethodCallHandler(this)
+    MobileCore.setMessagingDelegate(FlutterAEPMessagingDelegate(messageCache, channel))
+  }
+
+  override fun onMethodCall(call: MethodCall, result: Result) {
+    when (call.method) {
+      // Messaging Methods
+      "extensionVersion" -> result.success(Messaging.extensionVersion())
+      "getCachedMessages" -> this.getCachedMessages(result)
+      "refreshInAppMessages" -> result.success(Messaging.refreshInAppMessages())
+      // Message Methods
+      "clearMessage" -> this.clearMessage(call, result)
+      "dismiss" -> this.dismissMessage(call, result)
+      "handleJavascriptMessage" -> this.handleJavascriptMessage(call, result)
+      "setAutoTrack" -> this.setAutoTrack(call, result)
+      "showMessage" -> this.showMessage(call, result)
+      "trackMessage" -> this.trackMessage(call, result)
+      else -> {
+        result.notImplemented()
+      }
+    }
+  }
+
+  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
+    channel.setMethodCallHandler(null)
+  }
+
+  private fun getCachedMessages(result: Result) {
+    val cachedMessages = messageCache.values.map {
+        message -> mapOf("id" to message.id, "autoTrack" to true)
+    }.toList()
+    return result.success(cachedMessages)
+  }
+
+  // Message Class Functions
```

Comment:
Null or empty check for messageCache in each method, so we don't get error
E/flutter: [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: RangeError (index): Invalid value: Valid value range is empty: 0